# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1241728983 14400
#      Thu May 07 16:43:03 2009 -0400
# Node ID a3237ae32c0d1b6133b8188abc37dbfcd2ae7e8f
# Parent  49a0c58a7dcff88456468915e300d35fce106257
eliminate last remnants of F2C variables

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,13 @@
+2009-05-07  John W. Eaton  <jwe@octave.org>
+
+	* configure.in, mkoctfile.in, mkoctfile.cc.in, octave-bug.in,
+	octave-bug.cc.in: Eliminate last remnants of f2c goop.
+
 2009-05-07  Marco Atzeri  <marco_atzeri@yahoo.it>
 
 	* configure.in, Makeconf.in: (SHLPRE): Rename from SHLLIBPRE.
 
 2009-05-05  Jaroslav Hajek  <highegg@gmail.com>
 
 	* configure.in: Include the whole 3.x g++ series as problematic.
 	* Makefile: Don't mention that g++-3.x might work.
diff --git a/Makeconf.in b/Makeconf.in
--- a/Makeconf.in
+++ b/Makeconf.in
@@ -99,21 +99,16 @@ SHLBIN = @SHLBIN@
 SHLBIN_VER = @SHLBIN_VER@
 SHLLINKEXT=
 
 LIBEXT = a
 LIBPRE = @LIBPRE@
 SHLPRE = @SHLPRE@
 SHLBINPRE = @SHLBINPRE@
 
-# Fortran to C translator and associated flags.
-
-F2C = @F2C@
-F2CFLAGS = @F2CFLAGS@
-
 # Fortran compiler flags.
 
 FC = @FC@
 F77 = @F77@
 FFLAGS = @FFLAGS@
 FPICFLAG = @FPICFLAG@
 ALL_FFLAGS = $(FFLAGS)
 F77_FLOAT_STORE_FLAG = @F77_FLOAT_STORE_FLAG@
@@ -494,18 +489,16 @@ echo "making $@ from $<"
   -e "s|%OCTAVE_CONF_CXXPICFLAG%|\"${CXXPICFLAG}\"|" \
   -e "s|%OCTAVE_CONF_CXX_VERSION%|\"${CXX_VERSION}\"|" \
   -e "s|%OCTAVE_CONF_DEFAULT_PAGER%|\"${DEFAULT_PAGER}\"|" \
   -e "s|%OCTAVE_CONF_DEPEND_FLAGS%|\"${DEPEND_FLAGS}\"|" \
   -e "s|%OCTAVE_CONF_DEPEND_EXTRA_SED_PATTERN%|\"${DEPEND_EXTRA_SED_PATTERN}\"|" \
   -e "s|%OCTAVE_CONF_DL_LD%|\"${DL_LD}\"|" \
   -e "s|%OCTAVE_CONF_DL_LDFLAGS%|\"${DL_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_EXEEXT%|\"${EXEEXT}\"|" \
-  -e "s|%OCTAVE_CONF_F2C%|\"${F2C}\"|" \
-  -e "s|%OCTAVE_CONF_F2CFLAGS%|\"${F2CFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_F77%|\"${F77}\"|" \
   -e "s|%OCTAVE_CONF_F77_FLOAT_STORE_FLAG%|\"${F77_FLOAT_STORE_FLAG}\"|" \
   -e "s|%OCTAVE_CONF_FC%|\"${FC}\"|" \
   -e "s|%OCTAVE_CONF_FFLAGS%|\"${FFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_FFTW_LIBS%|\"${FFTW_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_FLIBS%|\"${FLIBS}\"|" \
   -e "s|%OCTAVE_CONF_FPICFLAG%|\"${FPICFLAG}\"|" \
   -e "s|%OCTAVE_CONF_GLPK_LIBS%|\"${GLPK_LIBS}\"|" \
diff --git a/configure.in b/configure.in
--- a/configure.in
+++ b/configure.in
@@ -2053,21 +2053,16 @@ AH_BOTTOM([
 #define X_CAST(T, E) (T) (E)
 
 #if defined (CXX_BROKEN_REINTERPRET_CAST)
 #define FCN_PTR_CAST(T, E) (T) (E)
 #else
 #define FCN_PTR_CAST(T, E) reinterpret_cast<T> (E)
 #endif
 
-#if defined(HAVE_F2C) && !defined(F77_FUNC)
-#  define F77_FUNC(x,X) x ## _
-#  define F77_FUNC_(x,X) x ## __
-#endif
-
 #if !defined(HAVE_DEV_T)
 typedef short dev_t;
 #endif
 
 #if !defined(HAVE_INO_T)
 typedef unsigned long ino_t;
 #endif
 
@@ -2170,31 +2165,25 @@ AC_CONFIG_FILES([octMakefile Makeconf te
   libcruft/ranlib/Makefile libcruft/slatec-fn/Makefile
   libcruft/slatec-err/Makefile libcruft/villad/Makefile
   libcruft/blas-xtra/Makefile libcruft/lapack-xtra/Makefile])
 
 AC_OUTPUT
 
 ### Print a summary so that important information isn't missed.
 
-if test -z "$F77"; then
-  FORT="$F2C $F2CFLAGS"
-else
-  FORT="$F77 $FFLAGS"
-fi
-
 AC_MSG_NOTICE([
 
 Octave is now configured for $canonical_host_type
 
   Source directory:     $srcdir
   Installation prefix:  $prefix
   C compiler:           $CC $XTRA_CFLAGS $WARN_CFLAGS $CFLAGS
   C++ compiler:         $CXX $XTRA_CXXFLAGS $WARN_CXXFLAGS $CXXFLAGS
-  Fortran compiler:     $FORT
+  Fortran compiler:     $F77 $FFLAGS
   Fortran libraries:    $FLIBS
   BLAS libraries:       $BLAS_LIBS
   FFTW libraries:       $FFTW_LIBS
   GLPK libraries:       $GLPK_LIBS
   UMFPACK libraries:    $UMFPACK_LIBS
   AMD libraries:        $AMD_LIBS
   CAMD libraries:       $CAMD_LIBS
   COLAMD libraries:     $COLAMD_LIBS
diff --git a/mkoctfile.cc.in b/mkoctfile.cc.in
--- a/mkoctfile.cc.in
+++ b/mkoctfile.cc.in
@@ -194,18 +194,16 @@ initialize (void)
     DEFAULT_INCFLAGS += " -I" + quote_path (vars["INCLUDEDIR"]);
 
   std::string DEFAULT_LFLAGS = "-L" + quote_path (vars["OCTLIBDIR"]);
   if (vars["LIBDIR"] != "/usr/lib")
     DEFAULT_LFLAGS += " -L" + quote_path (vars["LIBDIR"]);
 
   vars["CPPFLAGS"] = get_variable ("CPPFLAGS", %OCTAVE_CONF_CPPFLAGS%);
   vars["INCFLAGS"] = get_variable ("INCFLAGS", DEFAULT_INCFLAGS);
-  vars["F2C"] = get_variable ("F2C", %OCTAVE_CONF_F2C%);
-  vars["F2CFLAGS"] = get_variable ("F2CFLAGS", %OCTAVE_CONF_F2CFLAGS%);
   vars["F77"] = get_variable ("F77", %OCTAVE_CONF_F77%);
   vars["FFLAGS"] = get_variable ("FFLAGS", %OCTAVE_CONF_FFLAGS%);
   vars["FPICFLAG"] = get_variable ("FPICFLAG", %OCTAVE_CONF_FPICFLAG%);
   vars["CC"] = get_variable ("CC", %OCTAVE_CONF_CC%);
   vars["CFLAGS"] = get_variable ("CFLAGS", %OCTAVE_CONF_CFLAGS%);
   vars["CPICFLAG"] = get_variable ("CPICFLAG", %OCTAVE_CONF_CPICFLAG%);
   vars["CXX"] = get_variable ("CXX", %OCTAVE_CONF_CXX%);
   vars["CXXFLAGS"] = get_variable ("CXXFLAGS", %OCTAVE_CONF_CXXFLAGS%);
@@ -283,36 +281,35 @@ static string help_msg =
 "                          (or .mex if --mex is specified) unless linking\n"
 "                          a stand-alone executable.\n"
 "\n"
 "  -g                      Enable debugging options for compilers.\n"
 "\n"
 "  -p VAR, --print VAR     Print configuration variable VAR.  Recognized\n"
 "                          variables are:\n"
 "\n"
-"			    ALL_CFLAGS                FFTW_LIBS     \n"
-"			    ALL_CXXFLAGS              FLIBS       \n"
-"			    ALL_FFLAGS                FPICFLAG      \n"
-"			    ALL_LDFLAGS               INCFLAGS      \n"
-"			    BLAS_LIBS                 LDFLAGS             \n"
-"			    CC                        LD_CXX              \n"
-"			    CFLAGS                    LD_STATIC_FLAG\n"
-"			    CPICFLAG                  LFLAGS              \n"
-"			    CPPFLAGS                  LIBCRUFT      \n"
-"			    CXX                       LIBOCTAVE     \n"
-"			    CXXFLAGS                  LIBOCTINTERP  \n"
-"			    CXXPICFLAG                LIBREADLINE   \n"
-"			    DEPEND_EXTRA_SED_PATTERN  LIBS        \n"
-"			    DEPEND_FLAGS              OCTAVE_LIBS   \n"
-"			    DL_LD                     RDYNAMIC_FLAG \n"
-"			    DL_LDFLAGS                RLD_FLAG      \n"
-"			    F2C                       SED         \n"
-"			    F2CFLAGS                  XTRA_CFLAGS   \n"
-"			    F77                       XTRA_CXXFLAGS \n"
-"			    FFLAGS\n"
+"			    ALL_CFLAGS                FLIBS\n"
+"			    ALL_CXXFLAGS              FPICFLAG\n"
+"			    ALL_FFLAGS                INCFLAGS\n"
+"			    ALL_LDFLAGS               LDFLAGS\n"
+"			    BLAS_LIBS                 LD_CXX\n"
+"			    CC                        LD_STATIC_FLAG\n"
+"			    CFLAGS                    LFLAGS\n"
+"			    CPICFLAG                  LIBCRUFT\n"
+"			    CPPFLAGS                  LIBOCTAVE\n"
+"			    CXX                       LIBOCTINTERP\n"
+"			    CXXFLAGS                  LIBREADLINE\n"
+"			    CXXPICFLAG                LIBS\n"
+"			    DEPEND_EXTRA_SED_PATTERN  OCTAVE_LIBS\n"
+"			    DEPEND_FLAGS              RDYNAMIC_FLAG\n"
+"			    DL_LD                     RLD_FLAG\n"
+"			    DL_LDFLAGS                SED\n"
+"			    F77                       XTRA_CFLAGS\n"
+"			    FFLAGS                    XTRA_CXXFLAGS\n"
+"			    FFTW_LIBS\n"
 "\n"
 "  --link-stand-alone      Link a stand-alone executable file.\n"
 "\n"
 "  --mex                   Assume we are creating a MEX file.  Set the\n"
 "                          default output extension to \".mex\".\n"
 "\n"
 "  -s, --strip             Strip output file.\n"
 "\n"
@@ -627,25 +624,17 @@ main (int argc, char **argv)
 	}
 
       return 0;
     }
 
   for (it = f77files.begin (); it != f77files.end (); ++it)
     {
       string f = *it, b = basename (f);
-      if (!vars["F2C"].empty () && vars["F2C"] != "@F2C@")
-	{
-	  string c = b + ".c";
-	  cfiles.push_back (c);
-	  string cmd = vars["F2C"] + " " + vars["F2CFLAGS"]
-	    + " < " + f + " > " + c;
-	  result = run_command (cmd);
-	}
-      else if (!vars["F77"].empty ())
+      if (!vars["F77"].empty ())
 	{
 	  string o;
 	  if (!outputfile.empty ())
 	    {
 	      if (link)
 		o = b + ".o";
 	      else
 		o = outputfile;
diff --git a/mkoctfile.in b/mkoctfile.in
--- a/mkoctfile.in
+++ b/mkoctfile.in
@@ -2,27 +2,27 @@
 ##
 ## mkoctfile -- create a .oct file suitable for dynamic linking by
 ## Octave.
 ##
 ## Copyright (C) 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2004, 2005,
 ##               2006, 2007 John W. Eaton
 ##
 ## This file is part of Octave.
-## 
+##
 ## Octave is free software; you can redistribute it and/or modify it
 ## under the terms of the GNU General Public License as published by the
 ## Free Software Foundation; either version 3 of the License, or (at
 ## your option) any later version.
-## 
+##
 ## Octave is distributed in the hope that it will be useful, but WITHOUT
 ## ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 ## FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 ## for more details.
-## 
+##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 # Exit immediately on any error.
 
 set -e
 
@@ -57,24 +57,22 @@ if [ "$INCLUDEDIR" != /usr/include ]; th
 fi
 
 DEFAULT_LFLAGS="-L$OCTLIBDIR"
 if [ "$LIBDIR" != /usr/lib ]; then
   DEFAULT_LFLAGS="$DEFAULT_LFLAGS -L$LIBDIR"
 fi
 
 # Default values for these variables are filled in when Octave is
-# compiled. 
+# compiled.
 
 : ${EXEEXT=%OCTAVE_CONF_EXEEXT%}
 
 : ${CPPFLAGS=%OCTAVE_CONF_CPPFLAGS%}
 : ${INCFLAGS=$DEFAULT_INCFLAGS}
-: ${F2C=%OCTAVE_CONF_F2C%}
-: ${F2CFLAGS=%OCTAVE_CONF_F2CFLAGS%}
 : ${F77=%OCTAVE_CONF_F77%}
 : ${FFLAGS=%OCTAVE_CONF_FFLAGS%}
 : ${FPICFLAG=%OCTAVE_CONF_FPICFLAG%}
 : ${CC=%OCTAVE_CONF_CC%}
 : ${CFLAGS=%OCTAVE_CONF_CFLAGS%}
 : ${CPICFLAG=%OCTAVE_CONF_CPICFLAG%}
 : ${CXX=%OCTAVE_CONF_CXX%}
 : ${CXXFLAGS=%OCTAVE_CONF_CXXFLAGS%}
@@ -213,36 +211,35 @@ Options:
                           (or .mex if --mex is specified) unless linking
                           a stand-alone executable.
 
   -g                      Enable debugging options for compilers.
 
   -p VAR, --print VAR     Print configuration variable VAR.  Recognized
                           variables are:
 
-			    ALL_CFLAGS                FFTW_LIBS     
-			    ALL_CXXFLAGS              FLIBS       
-			    ALL_FFLAGS                FPICFLAG      
-			    ALL_LDFLAGS               INCFLAGS      
-			    BLAS_LIBS                 LDFLAGS             
-			    CC                        LD_CXX              
-			    CFLAGS                    LD_STATIC_FLAG
-			    CPICFLAG                  LFLAGS              
-			    CPPFLAGS                  LIBCRUFT      
-			    CXX                       LIBOCTAVE     
-			    CXXFLAGS                  LIBOCTINTERP  
-			    CXXPICFLAG                LIBREADLINE   
-			    DEPEND_EXTRA_SED_PATTERN  LIBS        
-			    DEPEND_FLAGS              OCTAVE_LIBS   
-			    DL_LD                     RDYNAMIC_FLAG 
-			    DL_LDFLAGS                RLD_FLAG      
-			    F2C                       SED         
-			    F2CFLAGS                  XTRA_CFLAGS   
-			    F77                       XTRA_CXXFLAGS 
-			    FFLAGS
+			    ALL_CFLAGS                FLIBS
+			    ALL_CXXFLAGS              FPICFLAG
+			    ALL_FFLAGS                INCFLAGS
+			    ALL_LDFLAGS               LDFLAGS
+			    BLAS_LIBS                 LD_CXX
+			    CC                        LD_STATIC_FLAG
+			    CFLAGS                    LFLAGS
+			    CPICFLAG                  LIBCRUFT
+			    CPPFLAGS                  LIBOCTAVE
+			    CXX                       LIBOCTINTERP
+			    CXXFLAGS                  LIBREADLINE
+			    CXXPICFLAG                LIBS
+			    DEPEND_EXTRA_SED_PATTERN  OCTAVE_LIBS
+			    DEPEND_FLAGS              RDYNAMIC_FLAG
+			    DL_LD                     RLD_FLAG
+			    DL_LDFLAGS                SED
+			    F77                       XTRA_CFLAGS
+			    FFLAGS                    XTRA_CXXFLAGS
+			    FFTW_LIBS
 
   --link-stand-alone      Link a stand-alone executable file.
 
   --mex                   Assume we are creating a MEX file.  Set the
                           default output extension to ".mex".
 
   -s, --strip             Strip output file.
 
@@ -412,34 +409,28 @@ if [ -n "$f77files" ]; then
         b=`echo $f | $SED 's,\.f90$,,'`
       ;;
       *.F90)
         b=`echo $f | $SED 's,\.F90$,,'`
       ;;
     esac
     if [ -n "$F77" ]; then
       if [ -n "$outputfile" ]; then
-	if $link; then 
+	if $link; then
 	  o=$b.o
 	else
 	  o=$outputfile
 	fi
       else
         o=$b.o
       fi
       objfiles="$objfiles $o"
       cmd="$F77 -c $FPICFLAG $ALL_FFLAGS $incflags $defs $pass_on_options $f -o $o"
       $dbg $cmd
       eval $cmd
-    elif [ -n "$F2C" ]; then
-      c=$b.c
-      cfiles="$cfiles $c"
-      cmd="$F2C $F2CFLAGS < $f > $c"
-      $dbg $cmd
-      eval $cmd
     else
       echo "mkoctfile: no way to compile Fortran file $f" 1>&2
     fi
   done
 fi
 
 if [ -n "$cfiles" ]; then
   for f in $cfiles; do
diff --git a/octave-bug.cc.in b/octave-bug.cc.in
--- a/octave-bug.cc.in
+++ b/octave-bug.cc.in
@@ -78,18 +78,16 @@ main (int argc, char **argv)
   vars["config_optvars"] = %OCTAVE_CONF_config_opts%;
   vars["VERSION"] = %OCTAVE_CONF_VERSION%;
   vars["SED"] = %OCTAVE_CONF_SED%;
   vars["MACHINE"] = %OCTAVE_CONF_CANONICAL_HOST_TYPE%;
   vars["F77"] = %OCTAVE_CONF_F77%;
   vars["FFLAGS"] = %OCTAVE_CONF_FFLAGS%;
   vars["FPICFLAG"] = %OCTAVE_CONF_FPICFLAG%;
   vars["FLIBS"] = %OCTAVE_CONF_FLIBS%;
-  vars["F2C"] = %OCTAVE_CONF_F2C%;
-  vars["F2CFLAGS"] = %OCTAVE_CONF_F2CFLAGS%;
   vars["CPPFLAGS"] = %OCTAVE_CONF_CPPFLAGS%;
   vars["INCFLAGS"] = %OCTAVE_CONF_INCFLAGS%;
   vars["CC"] = %OCTAVE_CONF_CC%;
   vars["CC_VERSION"] = %OCTAVE_CONF_CC_VERSION%;
   vars["CFLAGS"] = %OCTAVE_CONF_CFLAGS%;
   vars["CPICFLAG"] = %OCTAVE_CONF_CPICFLAG%;
   vars["CXX"] = %OCTAVE_CONF_CXX%;
   vars["CXX_VERSION"] = %OCTAVE_CONF_CXX_VERSION%;
@@ -163,18 +161,16 @@ main (int argc, char **argv)
   os << endl;
   os << "Configuration (please do not edit this section):" << endl;
   os << "-----------------------------------------------" << endl;
   os << endl;
   os << "uname output:     " << vars["UN"] << endl;
   os << "configure opts:   " << vars["config_opts"] << endl;
   os << "Fortran compiler: " << vars["F77"] << endl;
   os << "FFLAGS:           " << vars["FFLAGS"] << endl;
-  os << "F2C:              " << vars["F2C"] << endl;
-  os << "F2CFLAGS:         " << vars["F2CFLAGS"] << endl;
   os << "FLIBS:            " << vars["FLIBS"] << endl;
   os << "CPPFLAGS:         " << vars["CPPFLAGS"] << endl;
   os << "INCFLAGS:         " << vars["INCFLAGS"] << endl;
   os << "C compiler:       " << vars["CC_AND_VERSION"] << endl;
   os << "CFLAGS:           " << vars["CFLAGS"] << endl;
   os << "CPICFLAG:         " << vars["CPICFLAG"] << endl;
   os << "C++ compiler:     " << vars["CXX_AND_VERSION"] << endl;
   os << "CXXFLAGS:         " << vars["CXXFLAGS"] << endl;
diff --git a/octave-bug.in b/octave-bug.in
--- a/octave-bug.in
+++ b/octave-bug.in
@@ -30,18 +30,16 @@
 config_opts=%OCTAVE_CONF_config_opts%
 VERSION=%OCTAVE_CONF_VERSION%
 SED=%OCTAVE_CONF_SED%
 MACHINE=%OCTAVE_CONF_CANONICAL_HOST_TYPE%
 F77=%OCTAVE_CONF_F77%
 FFLAGS=%OCTAVE_CONF_FFLAGS%
 FPICFLAG=%OCTAVE_CONF_FPICFLAG%
 FLIBS=%OCTAVE_CONF_FLIBS%
-F2C=%OCTAVE_CONF_F2C%
-F2CFLAGS=%OCTAVE_CONF_F2CFLAGS%
 CPPFLAGS=%OCTAVE_CONF_CPPFLAGS%
 INCFLAGS=%OCTAVE_CONF_INCFLAGS%
 CC=%OCTAVE_CONF_CC%
 CC_VERSION=%OCTAVE_CONF_CC_VERSION%
 CFLAGS=%OCTAVE_CONF_CFLAGS%
 CPICFLAG=%OCTAVE_CONF_CPICFLAG%
 CXX=%OCTAVE_CONF_CXX%
 CXX_VERSION=%OCTAVE_CONF_CXX_VERSION%
@@ -226,18 +224,16 @@ Fix:
 
 Configuration (please do not edit this section):
 -----------------------------------------------
 
 uname output:     $UN
 configure opts:   $config_opts
 Fortran compiler: $F77
 FFLAGS:           $FFLAGS
-F2C:              $F2C
-F2CFLAGS:         $F2CFLAGS
 FLIBS:            $FLIBS
 CPPFLAGS:         $CPPFLAGS
 INCFLAGS:         $INCFLAGS
 C compiler:       $CC_AND_VERSION
 CFLAGS:           $CFLAGS
 CPICFLAG:         $CPICFLAG
 C++ compiler:     $CXX_AND_VERSION
 CXXFLAGS:         $CXXFLAGS
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,15 @@
+2009-05-07  John W. Eaton  <jwe@octave.org>
+
+	* toplev.cc (octave_config_info): Remove F2C and F2CFLAGS from the
+	config infor struct.
+	* oct-conf.h.in (OCTAVE_CONF_F2CFLAGS, OCTAVE_CONF_F2C):
+	Delete definitions.
+
 2009-05-07  Marco Atzeri  <marco_atzeri@yahoo.it>
 
         * Makefile.in: (SHLPRE): Rename from SHLLIBPRE.
 
 2009-05-07  John W. Eaton  <jwe@octave.org>
 
 	* genprops.awk (emit_declarations): Emit decls for static
 	has_property functions.
diff --git a/src/oct-conf.h.in b/src/oct-conf.h.in
--- a/src/oct-conf.h.in
+++ b/src/oct-conf.h.in
@@ -116,24 +116,16 @@ along with Octave; see the file COPYING.
 #ifndef OCTAVE_CONF_DL_LDFLAGS
 #define OCTAVE_CONF_DL_LDFLAGS %OCTAVE_CONF_DL_LDFLAGS%
 #endif
 
 #ifndef OCTAVE_CONF_EXEEXT
 #define OCTAVE_CONF_EXEEXT %OCTAVE_CONF_EXEEXT%
 #endif
 
-#ifndef OCTAVE_CONF_F2CFLAGS
-#define OCTAVE_CONF_F2CFLAGS %OCTAVE_CONF_F2CFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_F2C
-#define OCTAVE_CONF_F2C %OCTAVE_CONF_F2C%
-#endif
-
 #ifndef OCTAVE_CONF_F77
 #define OCTAVE_CONF_F77 %OCTAVE_CONF_F77%
 #endif
 
 #ifndef OCTAVE_CONF_F77_FLOAT_STORE_FLAG
 #define OCTAVE_CONF_F77_FLOAT_STORE_FLAG %OCTAVE_CONF_F77_FLOAT_STORE_FLAG%
 #endif
 
diff --git a/src/toplev.cc b/src/toplev.cc
--- a/src/toplev.cc
+++ b/src/toplev.cc
@@ -1160,18 +1160,16 @@ specified option.\n\
       { false, "CXXPICFLAG", OCTAVE_CONF_CXXPICFLAG },
       { false, "CXX_VERSION", OCTAVE_CONF_CXX_VERSION },
       { false, "DEFAULT_PAGER", OCTAVE_DEFAULT_PAGER },
       { false, "DEFS", OCTAVE_CONF_DEFS },
       { false, "DL_LD", OCTAVE_CONF_DL_LD },
       { false, "DL_LDFLAGS", OCTAVE_CONF_DL_LDFLAGS },
       { false, "ENABLE_DYNAMIC_LINKING", OCTAVE_CONF_ENABLE_DYNAMIC_LINKING },
       { false, "EXEEXT", OCTAVE_CONF_EXEEXT },
-      { false, "F2C", OCTAVE_CONF_F2C },
-      { false, "F2CFLAGS", OCTAVE_CONF_F2CFLAGS },
       { false, "F77", OCTAVE_CONF_F77 },
       { false, "F77_FLOAT_STORE_FLAG", OCTAVE_CONF_F77_FLOAT_STORE_FLAG },
       { false, "FC", OCTAVE_CONF_FC },
       { false, "FFLAGS", OCTAVE_CONF_FFLAGS },
       { false, "FFTW_LIBS", OCTAVE_CONF_FFTW_LIBS },
       { false, "FLIBS", OCTAVE_CONF_FLIBS },
       { false, "FPICFLAG", OCTAVE_CONF_FPICFLAG },
       { false, "GLPK_LIBS", OCTAVE_CONF_GLPK_LIBS },
